File: app/src/main/java/com/soft/credit911/Splash/SplashActivity.java
package com.soft.credit911.Splash;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;

import com.soft.credit911.Landing.LandingActivity;
import com.soft.credit911.Login.LoginActivity;
import com.soft.credit911.Utils.AppPreference;
import com.soft.credit911.databinding.ActivitySplashBinding;

public class SplashActivity extends AppCompatActivity {
    private ActivitySplashBinding layoutBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        layoutBinding = ActivitySplashBinding.inflate(getLayoutInflater());
        View view = layoutBinding.getRoot();
        setContentView(view);

        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);

        layoutBinding.tvGetStartNow.setOnClickListener(v -> {
            Intent intent;
            if (new AppPreference(this).getIsLogin())
                intent = new Intent(this, LandingActivity.class);
            else
                intent = new Intent(this, LoginActivity.class);
            startActivity(intent);
        });

    }
}

File: app/src/main/java/com/soft/credit911/ui/Splash/SplashActivity.kt
package com.soft.credit911.ui.Splash

import android.content.Intent
import android.util.Log
import android.view.View
import android.widget.Toast
import com.google.android.gms.tasks.OnCompleteListener
import com.google.firebase.messaging.FirebaseMessaging
import com.ing.quiz.shared_prefrences.Prefs
import com.ing.quiz.shared_prefrences.SharedPreferencesName
import com.ing.quiz.ui.base_classes.SubBaseActivity
import com.soft.credit911.ui.dashboard.LandingActivity
import com.soft.credit911.ui.Login.LoginActivity
import com.soft.credit911.R
import com.soft.credit911.Utils.AppPreference
import kotlinx.android.synthetic.main.activity_splash.*

class SplashActivity : SubBaseActivity() {

    override fun getLayoutID(): Int {
       return R.layout.activity_splash
    }

    override fun onViewCreated() {
        FirebaseMessaging.getInstance().setAutoInitEnabled(true);
        tv_Get_start_Now.setOnClickListener { v: View? ->
            val intent: Intent
            intent = if (AppPreference(this).isLogin) Intent(
                this,
                LandingActivity::class.java
            ) else Intent(this, LoginActivity::class.java)
            startActivity(intent)
        }

        FirebaseMessaging.getInstance().token.addOnCompleteListener(OnCompleteListener { task ->
            if (!task.isSuccessful) {
                Log.e("TAG", "Fetching FCM registration token failed", task.exception)
                return@OnCompleteListener
            }
            val token = task.result
            Prefs.with(this).save(SharedPreferencesName.DEVICETOKEN,token.toString())
        })
    }
}